hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/TransactionProcessorTest.java

```diff
@@ -156,45 +279,140 @@ void ethCallHappyPathAsExpected() {
-                        eq(transaction),
-                        eq(worldUpdater),
-                        eq(context),
-                        eq(config),
-                        eq(EIP_1014_ADDRESS),
-                        eq(NON_SYSTEM_LONG_ZERO_ADDRESS),
-                        eq(CHARGING_RESULT.intrinsicGas())))
+                        transaction,
+                        worldUpdater,
+                        context,
+                        config,
+                        EIP_1014_ADDRESS,
+                        NON_SYSTEM_LONG_ZERO_ADDRESS,
+                        CHARGING_RESULT.intrinsicGas()))
                 .willReturn(initialFrame);
         given(frameRunner.runToCompletion(
                         eq(transaction.gasLimit()), eq(initialFrame), eq(tracer), any(), eq(contractCreationProcessor)))
                 .willReturn(SUCCESS_RESULT);
+        given(initialFrame.getSelfDestructs()).willReturn(Set.of(NON_SYSTEM_LONG_ZERO_ADDRESS));
 
-        final var result = subject.processTransaction(transaction, worldUpdater, context, tracer, config);
+        final var result =
+                subject.processTransaction(transaction, worldUpdater, () -> feesOnlyUpdater, context, tracer, config);
 
         inOrder.verify(senderAccount).incrementNonce();
         inOrder.verify(gasCharging).chargeForGas(senderAccount, relayerAccount, context, worldUpdater, transaction);
         inOrder.verify(frameBuilder)
                 .buildInitialFrameWith(
-                        eq(transaction),
-                        eq(worldUpdater),
-                        eq(context),
-                        eq(config),
-                        eq(EIP_1014_ADDRESS),
-                        eq(NON_SYSTEM_LONG_ZERO_ADDRESS),
-                        eq(CHARGING_RESULT.intrinsicGas()));
+                        transaction,
+                        worldUpdater,
+                        context,
+                        config,
+                        EIP_1014_ADDRESS,
+                        NON_SYSTEM_LONG_ZERO_ADDRESS,
+                        CHARGING_RESULT.intrinsicGas());
         inOrder.verify(frameRunner)
                 .runToCompletion(
                         transaction.gasLimit(), initialFrame, tracer, messageCallProcessor, contractCreationProcessor);
+        inOrder.verify(gasCharging)
+                .maybeRefundGiven(
+                        GAS_LIMIT - SUCCESS_RESULT.gasUsed(),
+                        CHARGING_RESULT.relayerAllowanceUsed(),
+                        senderAccount,
+                        relayerAccount,
+                        context,
+                        worldUpdater);
+        inOrder.verify(worldUpdater).deleteAccount(NON_SYSTEM_LONG_ZERO_ADDRESS);
+        inOrder.verify(worldUpdater).commit();
         assertSame(SUCCESS_RESULT, result);
     }
 
+    @Test
+    @SuppressWarnings("unchecked")
+    void ethCallAsExpectedWithResourceExhaustionInCommit() {
```

Comment:
I don't understand the name exactly - does it mean that the eth call goes just fine up until the point where it's commit and then finally all of the sudden at the commit you run out of resources (in this case you short the renewal fees)?